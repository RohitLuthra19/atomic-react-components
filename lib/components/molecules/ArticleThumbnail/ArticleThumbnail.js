// @flow
/**
 *
 * ArticleThumbnail
 *
 */
import React from 'react';
import styled from 'styled-components';

import styles from './ArticleThumbnail.style';
import type { Props } from './types';
import Image from '../../atoms/Image/Image';
import Anchor from '../../atoms/Anchor';

const getAnchoredChild = (anchorProps, children) => <Anchor {...anchorProps}>{children}</Anchor>;

const ArticleThumbnail = ({ image, readMoreContent, children, anchorProps }: Props): Node => {
  let thumbImage = <Image {...image} />;

  if (readMoreContent && anchorProps) {
    thumbImage = getAnchoredChild(anchorProps, thumbImage);
  }
  if (readMoreContent) {
    const readMoreLink = getAnchoredChild(anchorProps, readMoreContent);
    return [thumbImage, children, readMoreLink];
  }

  return getAnchoredChild(anchorProps, [thumbImage, children]);
};

ArticleThumbnail.defaultProps = {
  image: {},
};

export default styled(ArticleThumbnail)`
  ${styles};
`;

export { ArticleThumbnail as ArticleThumbnailVanilla };
