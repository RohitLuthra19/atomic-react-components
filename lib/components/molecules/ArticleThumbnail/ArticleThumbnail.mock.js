const defaultImage = {
  src: '/images/dummies/320.png',
  srcset:
    '/images/dummies/1280.png 1024w, /images/dummies/640.png 640w, /images/dummies/320.png 320w',
  sizes: '(min-width: 36em) 33.3vw, 100vw',
  placeholderSrc: '/images/image-placeholder.png',
  alt: 'Dummy Image',
};

const anchor = {
  to: '#',
  className: 'AnchorClass',
  target: '_blank',
};

const defaultConfig = {
  image: { ...defaultImage },
  anchorProps: anchor,
};

const configV2 = {
  ...defaultConfig,
  image: { ...defaultConfig.image, anchorProps: anchor },
  anchorProps: anchor,
  readMoreContent: 'ReadMore',
};

const configV3 = { ...defaultConfig };
configV3.readMore = anchor;
configV3.readMore.children = 'Readmore';
export { configV2, configV3 };
export default defaultConfig;
