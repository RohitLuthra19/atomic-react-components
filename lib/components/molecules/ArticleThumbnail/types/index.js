// @flow

type anchor = {
  to: string | any,
  as?: string,
  children: Node,
  className?: string,
  inheritedStyles?: string,
  handleLinkClick?: (SyntheticEvent<>) => void,
  shallow?: boolean,
  scroll?: boolean,
  target?: '' | '_self' | '_blank' | '_parent' | '_top',
  forceClick?: boolean,
};

type image = {
  anchorPorps?: anchor,
  className: string,
  src: string,
  srcset?: string,
  sizes?: string,
  placeholderSrc?: string,
  alt: string,
  inheritedStyles?: string,
  ref: () => void,
};

type title = {
  anchorPorps?: anchor,
  children: Node,
  className: string,
  HeadingType?: string,
  inheritedStyles?: string,
};

type description = {
  children: Node,
  className?: string | void,
  inheritedStyles?: string,
};

export type Props = {
  image: image,
  title: title,
  url?: string,
  description?: description,
  readMore?: anchor,
  target?: string,
};
