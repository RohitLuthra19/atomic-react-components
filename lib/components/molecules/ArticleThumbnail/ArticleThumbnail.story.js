import React from 'react';
import { storiesOf } from '@storybook/react';
// import { action } from '@storybook/addon-actions';
import defaultConfig, { configV2 } from './ArticleThumbnail.mock';

// Import Styled Component to showcase variations
import ArticleThumbnail, { ArticleThumbnailVanilla } from '.';
import Para from '../../atoms/Para';
import Heading from '../../atoms/Heading';

console.log(ArticleThumbnail);
// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('Molecules', module).addWithChapters('ArticleThumbnail', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => (
            <ArticleThumbnailVanilla {...defaultConfig} className={`${defaultConfig.className}`} />
          ),
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Primary ArticleThumbnail Variations',
      sections: [
        {
          title: 'Variation Name : With whole thumb clickable',
          sectionFn: () => <ArticleThumbnail {...defaultConfig} />,
        },
      ],
    },
    {
      title: 'Primary ArticleThumbnail Variations V2',
      sections: [
        {
          title: 'Variation Name : With Readmore, image clickable',
          sectionFn: () => (
            <ArticleThumbnail {...configV2}>
              <React.Fragment>
                <Heading>Dummy Title</Heading>
                <Para>Dummy Para</Para>
              </React.Fragment>
            </ArticleThumbnail>
          ),
        },
      ],
    },
  ],
});
