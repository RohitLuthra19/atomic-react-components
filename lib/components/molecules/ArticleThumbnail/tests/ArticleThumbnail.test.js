import { shallow } from 'enzyme';

import ArticleThumbnail from '../index';
import defaultConfig, { configV3, configV2 } from '../ArticleThumbnail.mock';

describe('<ArticleThumbnail />', () => {
  test('should render correctly', () => {
    const ArticleThumbnailComponent = shallow(<ArticleThumbnail>Test</ArticleThumbnail>);
    expect(ArticleThumbnailComponent).toMatchSnapshot();
  });
  test('should render correctly with readmore button', () => {
    const ArticleThumbnailComponent = shallow(
      <ArticleThumbnail {...configV3}>Test</ArticleThumbnail>
    );
    expect(ArticleThumbnailComponent).toMatchSnapshot();
  });
  test('should render correctly with clickable image and title', () => {
    const ArticleThumbnailComponent = shallow(
      <ArticleThumbnail {...configV2}>Test</ArticleThumbnail>
    );
    expect(ArticleThumbnailComponent).toMatchSnapshot();
  });
  test('should render correctly with no clickable image and title', () => {
    const ArticleThumbnailComponent = shallow(
      <ArticleThumbnail {...defaultConfig}>Test</ArticleThumbnail>
    );
    expect(ArticleThumbnailComponent).toMatchSnapshot();
  });
});
