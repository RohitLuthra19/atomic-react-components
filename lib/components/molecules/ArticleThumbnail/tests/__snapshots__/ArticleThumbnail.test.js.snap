// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleThumbnail /> should render correctly 1`] = `
<ArticleThumbnail
  className="ArticleThumbnail-sc-1pvkf9f-0 hCXvzE"
  image={Object {}}
>
  Test
</ArticleThumbnail>
`;

exports[`<ArticleThumbnail /> should render correctly with clickable image and title 1`] = `
<ArticleThumbnail
  anchorProps={
    Object {
      "className": "AnchorClass",
      "target": "_blank",
      "to": "#",
    }
  }
  className="ArticleThumbnail-sc-1pvkf9f-0 hCXvzE"
  image={
    Object {
      "alt": "Dummy Image",
      "placeholderSrc": "/images/image-placeholder.png",
      "sizes": "(min-width: 36em) 33.3vw, 100vw",
      "src": "/images/dummies/320.png",
      "srcset": "/images/dummies/1280.png 1024w, /images/dummies/640.png 640w, /images/dummies/320.png 320w",
    }
  }
  readMoreContent="ReadMore"
>
  Test
</ArticleThumbnail>
`;

exports[`<ArticleThumbnail /> should render correctly with no clickable image and title 1`] = `
<ArticleThumbnail
  anchorProps={
    Object {
      "className": "AnchorClass",
      "target": "_blank",
      "to": "#",
    }
  }
  className="ArticleThumbnail-sc-1pvkf9f-0 hCXvzE"
  image={
    Object {
      "alt": "Dummy Image",
      "placeholderSrc": "/images/image-placeholder.png",
      "sizes": "(min-width: 36em) 33.3vw, 100vw",
      "src": "/images/dummies/320.png",
      "srcset": "/images/dummies/1280.png 1024w, /images/dummies/640.png 640w, /images/dummies/320.png 320w",
    }
  }
>
  Test
</ArticleThumbnail>
`;

exports[`<ArticleThumbnail /> should render correctly with readmore button 1`] = `
<ArticleThumbnail
  className="ArticleThumbnail-sc-1pvkf9f-0 hCXvzE"
  image={Object {}}
>
  Test
</ArticleThumbnail>
`;
